src/preferences/dialog/dlgpreflv2.cpp

```diff
@@ -76,9 +78,8 @@ void DlgPrefLV2::slotDisplayParameters() {
     m_pluginParameters.clear();
 
     QLayoutItem* item;
-    while ((item = lv2_vertical_layout_params->takeAt(1)) != 0) {
-        lv2_vertical_layout_params->removeWidget(item->widget());
-        delete item->widget();
+    while ((item = lv2EffectParametersList->takeAt(0)) != 0) {
+        delete item;
```

Comment:
I'm not a fan of assignments in condition expressions. 

```suggestion
    while (! lv2EffectParametersList->isEmpty()) {
        QLayoutItem* item = lv2EffectParametersList->takeAt(0);
        VERIFY_OR_DEBUG_ASSERT(item) {
            continue;
        }
        delete item;
} 
```

Reply:
here `isEmpty()` doesn't count spacers which would remain at index 0 after having deleted all _real_ items.
`count()` does the trick :+1: